presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -27,49 +30,41 @@
 import static java.util.concurrent.Executors.newSingleThreadExecutor;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
+@ThreadSafe
 public final class CommandExecutor
 {
     private static final Logger LOGGER = Logger.get(CommandExecutor.class);
-    private static final int DEFAULT_TIMEOUT = 60;
-    private final String[] commandArray;
-    private final int timeout;
 
-    private CommandExecutor(String[] commandArray, int timeout)
-    {
-        if (commandArray.length == 0) {
-            throw new IllegalArgumentException("Command array is empty");
-        }
-        this.commandArray = commandArray;
-        this.timeout = timeout;
-    }
+    // TODO: Make CommandExecutor non-static, injecting the timeout, possibly via a factory
+    private final int longTimeout;
+    private final int shortTimeout;
 
-    public static int executeCommand(String... command)
-            throws PrestoManagerException
+    @Inject
+    private CommandExecutor(AgentConfig config)
     {
-        return new CommandExecutor(command, DEFAULT_TIMEOUT).execute().getExitValue();
+        this.longTimeout = config.getLongSubprocessTimeout();
+        this.shortTimeout = config.getShortSubprocessTimeout();
     }
 
-    public static CommandResult execCommandResult(String... command)
+    public CommandResult runCommand(String... command)
             throws PrestoManagerException
     {
-        return new CommandExecutor(command, DEFAULT_TIMEOUT).execute();
+        return execute(command, shortTimeout);
     }
 
-    public static int executeCommand(int timeoutInSeconds, String... command)
+    public CommandResult runLongCommand(String... command)
```

Comment:
You want a better name for this. It is not Long command, but rather Long timeout